### ISSUE TITLE
Provider hang, logs suggest a missing AzureRM Response

### ISSUE THREAD

**Issue by dimbleby** [terraform-providers/terraform-provider-azurerm#9662]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```
Terraform v0.13.5
+ provider registry.terraform.io/hashicorp/azurerm v2.36.0
+ provider registry.terraform.io/hashicorp/local v2.0.0
+ provider registry.terraform.io/hashicorp/random v3.0.0
```

### Affected Resource(s)

All

### Terraform Configuration Files

Will try to share a minimized version if it proves necessary, but I'm pretty sure this is largely irrelevant.

### Debug Output

https://gist.github.com/dimbleby/a8bf8ae544943ae4827226a18a90cf11

Note that there is an AzureRM Request with no corresponding response here, specifically this one:
```
2020-12-01T21:32:31.277Z [DEBUG] plugin.terraform-provider-azurerm_v2.36.0_x5: [DEBUG] AzureRM Request:
2020-12-01T21:32:31.277Z [DEBUG] plugin.terraform-provider-azurerm_v2.36.0_x5: GET /subscriptions/REDACTED/providers?api-version=2016-02-01 HTTP/1.1
2020-12-01T21:32:31.277Z [DEBUG] plugin.terraform-provider-azurerm_v2.36.0_x5: Host: management.azure.com
2020-12-01T21:32:31.277Z [DEBUG] plugin.terraform-provider-azurerm_v2.36.0_x5: User-Agent: Go/go1.14.5 (amd64-linux) go-autorest/v14.2.1 Azure-SDK-For-Go/v48.1.0 resources/2016-02-01 HashiCorp Terraform/0.13.5 (+https://www.terraform.io) Terraform Plugin SDK/1.13.1 terraform-provider-azurerm/2.36.0 pid-222c6c49-1b0a-5959-a213-6608f9eb8820
2020-12-01T21:32:31.277Z [DEBUG] plugin.terraform-provider-azurerm_v2.36.0_x5: X-Ms-Correlation-Request-Id: 9dd25861-e936-3c86-2893-db907d989b21
2020-12-01T21:32:31.277Z [DEBUG] plugin.terraform-provider-azurerm_v2.36.0_x5: Accept-Encoding: gzip
```

I have omitted the first 20,000 lines or so of TRACE output as it probably includes some secrets, but can provide privately if required.

After that we see the same few lines repeating every 5 seconds indefinitely: https://gist.github.com/dimbleby/c17726bfda86b877ad4622e94c096fc6.  Everything is waiting for `"provider[\"registry.terraform.io/hashicorp/azurerm\"]"`

### Panic Output

N/A

### Expected Behaviour

If something has gone wrong, terraform should fail and exit.  Else it should make progress.

### Actual Behaviour

Terraform loops indefinitely, making no progress.  We have seen this go on for several hours before getting bored and killing it.

It looks as though this provider has either failed to set a timeout on the outstanding request, or perhaps swallowed an error somewhere.

### Steps to Reproduce

1. `terraform apply`

This only happens occasionally.

### References

* #0000

**Comment by ArcturusZhang** [comment]
Hi @dimbleby thanks for this issue!

You are right, it is the missing return there that is causing the problem.
We will fix this in the next version of the SDK. Sorry for the inconvenience.

**Comment by tombuildsstuff** [comment]
👋 

Taking a look through here, based on what I can see this has been fixed in version 2.45.0 of the Azure Provider (however I'd recommend upgrading to v2.45.1 since that fixes a panic in App Service Environments) - since these include updates to the base library used by the Azure SDK. As such I'm going to close this issue for the moment, but if you're still seeing this issue with the new version please let us know and we'll take another look.

Thanks!
